### ISSUE TITLE
[web-platform] flutter service worker cache is not skipping requests to other sites

### ISSUE THREAD

**Issue by AoEiuV020** [flutter/flutter#103176]
I think a host judgment should be added here to bypass some restrictions in serviceWorker fetch listener,
```js
  if (event.request.host != self.location.host) {
      return;
  }
```
Otherwise, requesting other websites will also be recognized as resources in RESOURCES, trying to read the cache,
This can lead to some unknown issues, including #103054

![image](https://user-images.githubusercontent.com/11682198/167071662-b2ea1e35-78be-4372-a857-a4f36c508a6c.png)
![image](https://user-images.githubusercontent.com/11682198/167071758-55e12671-7f83-4283-84ad-6e8c0a03ef70.png)

@darshankawar 
@ditman

**Comment by darshankawar** [comment]
Keeping it open and labeling for further insights from the team.

**Comment by ditman** [comment]
Good call, this is a bug in the service worker. It is supposed to cache only local resources. It's clear by the caching logic here:

* https://github.com/flutter/flutter/blob/78ae72b0b264bf9b8699ceb0f36f298dbba74299/packages/flutter_tools/lib/src/build_system/targets/web.dart#L673

That is computing a cache key that makes no sense, unless `event.request.url` starts with `origin`.

**Comment by christopherfujino** [comment]
@ditman are you still looking at this one?

**Comment by ditman** [comment]
@christopherfujino not actively working on solving it right now. I'm going to remove my association to this bug report for now (pls leave the "flutter.js" label so I can find it later!)

**Comment by Instadex** [comment]
Any update on this issue? I am also facing the same Mixed content issue on flutter web.
